### PR TITLE
FestivalContext

### PR DESCRIPTION
**Pull request by dlamkins**
WIP: Added the FestivalContext which provides a list of the active festivals.

Biggest missing item is that Gw2Sharp does not currently support accessing the achievement/groups endpoint due to a bug (fixed here: https://github.com/Archomeda/Gw2Sharp/issues/22) and an inconsistency in the API (see: https://github.com/arenanet/api-cdi/issues/657).  We'll have to update our Gw2Sharp dependency (which will be its own mini project).

This PR aims to satisfy #123.

### CODE REVIEW

File: Blish HUD/GameServices/Contexts/FestivalContext.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            { 98,  Festival.Wintersday},
+            { 162, Festival.SuperAdventureFestival },
+            { 201, Festival.LunarNewYear },
+            { 213, Festival.FestivalOfTheFourWinds },
+            { 233, Festival.DragonBash }
+        };
+
+        private CancellationTokenSource _contextLoadCancellationTokenSource;
+
+        private List<Festival> _activeFestivals = new List<Festival>();
```
Comment (greaka) [inline Blish HUD/GameServices/Contexts/FestivalContext.cs:102]: while it is technically possible that multiple festivals are active at once, theres no way anet will ever do this. for one, the decorations in cities would probably clip and they also stated that festivals are fillers between content releases. so i think they will never activate multiple festivals at the same time. 

what are your thoughts on providing just a single `_activeFestival`?

Comment (dlamkins) [reply Blish HUD/GameServices/Contexts/FestivalContext.cs:102]: I do agree that is is particularly unlikely, but with the increased number of "events" happening lately, I worry that "festivals" may begin to include some smaller events that potentially overlap.  I also get that it's less convenient not being able to check for just an individual festival.  Do the two new functions provided in my latest commit make you feel better about this?

Comment (greaka) [reply Blish HUD/GameServices/Contexts/FestivalContext.cs:102]: better :)

**Review by greaka — DISMISSED** [review]

**Comment by dlamkins** [comment]
Currently waiting for #172

**Comment by dlamkins** [comment]
#172 has been merged so this is now ready once it is reviewed again.  :)

**Review by nickcode122 — APPROVED** [review]

**Pull request merged**